libs/ui-components/src/components/common/LeaveFormConfirmation.tsx

```diff
@@ -79,11 +79,16 @@ const BrowserBlocker = ({ lock }: { lock: boolean }) => {
 const LeaveFormConfirmation = () => {
   const { dirty, isSubmitting } = useFormikContext();
   const {
+    appType,
     router: { useBlocker },
   } = useAppContext();
 
   const lock = !isSubmitting && dirty;
 
+  if (appType === 'aap') {
```

Comment:
even though AAP UI uses new react-router, we cannot (yet) use the `useBlocker` hook as they still use `BrowserRouter` and not data router.